ff/serving/coordinator/coordinator.go

```diff
@@ -90,21 +112,120 @@ func (c *Coordinator) WatchForNewJobs() error {
 	return nil
 }
 
-func (c *Coordinator) runRegisterSourceJob(resID metadata.ResourceID) error {
-	//the idea is that we register a csv or a postgres table
-	//and via that provider we should be able to "get" any of those offline tables from here on out
-	//and if a feature declares a source in its definition, the training set job should be able to go
-	//and get the namevariant shit
+func (c *Coordinator) mapNameVariantsToTables(sources []metadata.NameVariant) (map[string]string, error) {
+	sourceMap := make(map[string]string)
+	for _, nameVariant := range sources {
+		var tableName string
+		source, err := c.Metadata.GetSourceVariant(context.Background(), nameVariant)
+		if err != nil {
+			return nil, err
+		}
+		if source.Status() != metadata.READY {
+			return nil, fmt.Errorf("source in query not ready")
+		}
+		providerResourceID := provider.ResourceID{Name: source.Name(), Variant: source.Variant()}
+		if source.IsSQLTransformation() {
+			tableName = provider.GetTransformationName(providerResourceID)
+		} else if source.IsPrimaryDataSQLTable() {
+			tableName = provider.GetPrimaryTableName(providerResourceID)
+		}
+		sourceMap[fmt.Sprintf("%s.%s", nameVariant.Name, nameVariant.Variant)] = tableName
```

Comment:
make a struct called name variant, and make the map between that type and a string. If you do it this way, you risk ambiguity with a dot in it.